app/models/payment_intent.rb

```diff
@@ -7,6 +7,8 @@ class PaymentIntent < ApplicationRecord
   belongs_to :confirmation_source, polymorphic: true, optional: true
   belongs_to :cancellation_source, polymorphic: true, optional: true
 
+  has_many :registration_payment
```

Comment:
Nitpick, but important for Rails AR: By convention, `has_many` names should be in plural. I think (but only 40% sure) that this will fail to resolve in its current form.